crates/sui/tests/shared_objects_version_tests.rs

```diff
@@ -96,40 +76,6 @@ async fn initial_shared_version_mismatch_current_version() {
     assert_eq!(location.module.address(), &SUI_FRAMEWORK_ADDRESS);
     assert_eq!(location.module.name().as_str(), "transfer");
     assert_eq!(code, 0 /* ESharedNonNewObject */);
-    // let (counter, _) = env
-    //     .increment_shared_counter(counter, counter.1)
-    //     .await
-    //     .unwrap();
-
-    // let fx = env.increment_shared_counter(counter, counter.1).await;
-    // let err = fx.expect_err("Transaction fails");
-    // assert!(
-    //     is_txn_input_error(&err, "SharedObjectStartingVersionMismatch"),
-    //     "{}",
-    //     err
-    // );
-}
-
-#[sim_test]
-async fn initial_shared_version_mismatch_arbitrary() {
```

Comment:
Thanks for the info @amnn. I see there are logic related to `initial_shared_version` in `record_shared_object_cert_from_consensus()`.

For this change, it makes certificates submitted to Sui via RPC that are missing shared input objects to wait until the input is available, instead of executing immediately and returning the `SharedObjectPriorVersionsPendingExecution` error. This should be more desirable, otherwise inefficient retry logic will be necessary in client or in Sui RPC handler that does not have visibility into when the prior version actually gets executed. So the test starting to hang instead of returning the error is expected.

But if the feature is currently used, I want to make sure at least there is a happy path test for how this feature is commonly used. Does the current test cases include the happy path?

Reply:
And just to make sure, there are logic to verify that `initial_shared_version` cannot be set to an "invalid" value when signing transactions, right?

Reply:
Thanks for the context @mwtian, and agreed that sounds super helpful.  I've added another inline comment on this PR at the location of the happy path test for initial shared version.  The test that we are looking at here (that was deleted) is the one to make sure that we cannot run a transaction with an incorrect initial shared version.

Reply:
Sounds good. AuthorityState and TransactionManager ensure certified transactions that depend on a not-yet-exist shared object does not run immediately, but run when the object  & version becomes available. This logic is tested in a few places. I think an unit test of `record_shared_object_cert_from_consensus()` or `process_consensus_transaction()` might be useful to make sure shared object locks are set to the correct `initial_shared_version`.

Reply:
I'm a little unclear on what the resolution of this discussion was - is there another test that is covering this case? If not, lets leave this test in and just verify that it times out within 5 seconds or so.